models/sipnet/R/write.configs.SIPNET.R

```diff
@@ -438,7 +448,31 @@ write.config.SIPNET <- function(defaults, trait.values, settings, run.id, inputs
     if ("leafGrowth" %in% pft.names) {
       param[which(param[, 1] == "leafGrowth"), 2] <- pft.traits[which(pft.names == "leafGrowth")]
     }
-  }  ## end loop over PFTS
+
+    #update LeafOnday and LeafOffDay
+     if (!is.null(settings$run$inputs$leaf_phenology)){
+     obs_year_start <- lubridate::year(settings$run$start.date)
+     obs_year_end <- lubridate::year(settings$run$end.date)
+     if (obs_year_start != obs_year_end) {
+      PEcAn.logger::logger.info("Start.date and end.date are not in the same year. Currently start.date is used for refering phenological data")
+     }
+     leaf_pheno_path <- settings$run$inputs$leaf_phenology$path  ## read from settings
+      if (!is.null(leaf_pheno_path)){
+    ##read data
+       leafphdata <- utils::read.csv(leaf_pheno_path)
+       leafOnDay <-  dplyr::filter(leafphdata, .data$yea r== obs_year_start & .data$site_id==settings$run$site$id) %>%  dplyr::select (.data$leafonday)
+       leafOffDay<-  dplyr::filter(leafphdata, .data$year == obs_year_start & .data$site_id==settings$run$site$id) %>%  dplyr::select (.data$leafoffday)
```

Comment:
typo fix + formatting while I was here:
```suggestion
       leafOnDay <-  dplyr::filter(leafphdata, .data$year == obs_year_start & .data$site_id==settings$run$site$id) %>%
         dplyr::select (.data$leafonday)
       leafOffDay<-  dplyr::filter(leafphdata, .data$year == obs_year_start & .data$site_id==settings$run$site$id) %>%
         dplyr::select (.data$leafoffday)
```